JustSaying/AwsTools/MessageHandling/SnsTopicBase.cs

```diff
@@ -68,17 +72,21 @@ public void Publish(Message message)
                         $"Failed to publish message to SNS. TopicArn: {request.TopicArn} Subject: {request.Subject} Message: {request.Message}",
                         ex);
             }
+            finally
+            {
+                _messageResultLogger?.Invoke(new MessageResult(response?.MessageId, response?.HttpStatusCode));
```

Comment:
Is it that useful to call the delegate if all the properties of the `MessageResult` are `null`?

Reply:
I think so personally, call the delegate regardless of response for a couple of reasons:
- Principle of least surprise (As a consumer I'd expect the handler to be invoked regardless of response).
- Reduces complexity, far simpler API

Reply:
Would it be better to extend it to optionally also provide the exception? Otherwise getting an object with null properties just means "something went wrong" without telling you what. Then it's always called and always has something useful to tell the callee.

Reply:
I think that's a good idea

Reply:
Actually, on further thought I think we go with your initial comment as we already have an exception handler. I'll move this call out of the `finally` block and into the `try` one.